### PR TITLE
scalar x matrix (4x4) using NEON SIMD intrinsics

### PR DESCRIPTION
**Pull request by gMerm**

### CODE REVIEW

File: scalarxmat44_neon.c [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+  int are_equal = 1;
+  float lamda = 5.0;
+  int count = 0;
+
+  // alocate alligned memory for both A,B matrices
+  //(16 byte allignment because of NEON intrinsics)
+  float *A_memory = NULL;
+  float *B_memory = NULL;
+  if (posix_memalign((void **)&A_memory, 16, sizeof(float[N][N])) != 0) {
```
Comment (markos) [inline scalarxmat44_neon.c]: same comment as https://github.com/VectorCamp/simd-examples/pull/6#discussion_r1526480211
no need for dynamic allocation in the case of so small matrices.

Comment (gMerm) [reply scalarxmat44_neon.c]: I added it because in bigger matrices it would be of great help.
Should I remove it ?

Comment (markos) [reply scalarxmat44_neon.c]: yes please, keep it simple for now

**Comment by gMerm** [comment]
Scalar: 27 sec
NEON: 17 sec

**Comment by markos** [comment]
missing changes to Makefile, also Makefile has weird tabs.

File: scalarxmat44_neon.c [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+
+#define N 4
+#define LOOPS 10
+
+void scalarxmat44(float A[N][N], float lamda) {
```
Comment (markos) [inline scalarxmat44_neon.c]: as suggested elsewhere, please rename the C version as scalarxmat44_c.

File: scalarxmat44_neon.c [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    }
+  }
+}
+
+void scalarxmat44_neon(float32x4_t lamda_vector, float32x4_t array_in,
+                       float *B[]) {
+  for (int i = 0; i < 4; i++) {
+    // load 4 float values from the array B, multiply each value with lamda
+    // and then store result back to B array
+    array_in = vld1q_f32(&B[i][0]);
```
Comment (markos) [inline scalarxmat44_neon.c]: you are not using the passed argument array_in, you could just declare it directly here.

Comment (markos) [reply scalarxmat44_neon.c]: and remove the function parameter.

Comment (gMerm) [reply scalarxmat44_neon.c]: didn't do it this way because it would get declared N times, but this way it's only declared once

**Review by markos — APPROVED** [review]
LGTM, Approved

**Pull request merged**